photon-ml/src/main/scala/com/linkedin/photon/ml/Driver.scala

```diff
@@ -169,19 +169,26 @@ protected[ml] class Driver(
   }
 
   protected def prepareTrainingData(): Unit = {
+    // Verify that selected features files exist
     params.selectedFeaturesFile.foreach(file => {
       val path = new Path(file)
       if (! path.getFileSystem(sc.hadoopConfiguration).exists(path)) {
         throw new IOException("Could not find [" + file + "]. Check that the file exists")
       }
     })
+
+    // Read training data
     trainingData = suite.readLabeledPointsFromAvro(
       sc, params.trainDir, params.selectedFeaturesFile, params.minNumPartitions)
       .persist(trainDataStorageLevel).setName("training data")
-    featureNum = trainingData.first().features.size
+
     trainingDataNum = trainingData.count().toInt
+    require(trainingDataNum > 0,
+      "No training data found. Ensure that training data exists and feature vectors are not empty.")
```

Comment:
That's right, @XianXing. We map over the records and return ```None``` if the feature vector is empty, and then we ```flatMap``` over the result. This means that a dataset of empty feature vectors becomes an empty collection.